koku/koku/settings.py

```diff
@@ -143,6 +143,7 @@
 
 DEVELOPMENT = ENVIRONMENT.bool("DEVELOPMENT", default=False)
 if DEVELOPMENT:
+    DEVELOPMENT_IDENTITY = ENVIRONMENT.json("DEVELOPMENT_IDENTITY", default={})
```

Comment:
What happens if we hit the default?
Do we the default to be https://github.com/project-koku/koku/pull/2113/files#diff-86a0cdc27a1ea2f22801c8f80dbaea42R62-R68
and then we don't have an if/else?

Reply:
There's two if/else conditions, one in `settings.py` and one in `dev_middleware.py`

In `settings.py`, the logic is, "if `DEVELOPMENT` is set, look for the `DEVELOPMENT_IDENTITY`. This way, there's no chance to sneak a forged identity into the app without setting the `DEVELOPMENT` env var.

In `dev_middleware.py`, the logic is, "if `DEVELOPMENT_IDENTITY` is True (empty dicts are False), then use the forged identity."

Does this better explain the behavior?  Do you want it to do something else, instead?

Reply:
Sorry. Maybe I wasn't clear if some one didn't update their `.env` file with the `DEVELOPMENT_IDENTITY` you showed the `.env.example` file the `ENVIRONMENT.json()` method will take `{}` as the default. Feels like our code would break. 

AIso with this change I don't think we can make it down the `else` path in `dev_middleware.py`, since you always set a value when `DEVELOPMENT` is set.

So I was suggesting why not make `default={}` --> `default=[what's in the .env.example]` then you can delete the else condition from the `dev_middleware.py` Additionally no one needs to update their `.env` unless they want to take advantage of the RBAC override feature.

Reply:
I believe I addressed that in my response. In Python, empty dicts == False.

For example:
```
ipython
/usr/lib/python3.7/site-packages/IPython/core/interactiveshell.py:925: UserWarning: Attempting to work in a virtualenv. If you encounter problems, please install IPython inside the virtualenv.
  warn("Attempting to work in a virtualenv. If you encounter problems, please "
Python 3.7.7 (default, Mar 13 2020, 10:23:39) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.11.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: bool({})                                                                                                                                            
Out[1]: False
```

So, if we take the default value of `{}` in from the first conditional, it is a False value for the second conditional, which means we revert to the old behavior.

Am I missing something there?

Reply:
My apologies @blentz , too much multi-tasking and I read right past the "(empty dicts are False)" without comprehending.

To your point:

> So, what I need here is less about the exact mechanics of how this particular code can be made better and more an answer to the question of should we do it this way at all?

Yes, I'm in favor of this as a mechanism.